### PR TITLE
Update all links to use `docs/reference` style

### PR DESCRIPTION
**Pull request by jdbaldry**
# What this PR does

The `docs/reference` shortcode supports contextual destinations and version inference.

`<ONCALL VERSION>` is inferred to match the version of the documentation set. For example, the inferred version for the page /docs/grafana/oncall/latest/get-started/ is "latest". It can also be overriden using front matter.
Given the same page, but with the additional front matter `oncall_version: next`, the variable is substituted with "next" rather than "latest".

Contextual destinations are achieved using repeated labels in the shortcode inner text. The format is [<LABEL>]: "<PAGE PATH PREFIX> -> <HUGO REFERENCE>".

- _`<LABEL>`_ matches the reference style link label used in the rest of the text.
- _`<PAGE PATH PREFIX>`_ is matched against the page during the production build. If the match is successful, the destination that is used is _`<HUGO REFERENCE>`_. The first matching prefix is used, not the longest matching prefix.

## Which issue(s) this PR fixes

- Broken links due to ambiguous relref resolution. Any relref parameter that does not start with either `/`, `./`, or `../` can resolve ambiguously and is resulting in broken link behavior on the current site.
- Broken links in Grafana Cloud. We mount OnCall documentation in Grafana Cloud. In https://github.com/grafana/website/pull/13872 the location will become /docs/grafana-cloud/alerting-and-irm/oncall. This PR is intended to be merged alongside that PR.

### CODE REVIEW

**Comment by jdbaldry** [comment]
@grafana/grafana-oncall-backend would somebody be able to give me some pointers on how to address the want of `markdownlint` to remove the second reference labels?  

I already disable the inline HTML rule because the parser thinks `<ONCALL VERSION>` is inline HTML when it is not.

Possibly, disabling all the rules for that section of the file will work as the inner text is not valid Markdown until Hugo has processed the shortcode.

**Comment by joeyorlando** [comment]
@jdbaldry 

it looks like `markdownlint` thinks some of the links you added, are links to images ([example](https://github.com/grafana/oncall/pull/2456/files#diff-eca2c4923e552e7141ebc0b51774e335f65a4ba38af9aaa3c369869014fc2423R29)). From your [most recent build](https://github.com/grafana/oncall/actions/runs/5479881777/jobs/9982320210?pr=2456#step:7:127):
```bash
./docs/sources/escalation-chains-and-routes/_index.md:14:18 MD052/reference-links-images Reference links and images should use a label that is defined [Missing link or image reference definition: "routing-template"] [Context: "["Routing Templates"][routing-template]"]
./docs/sources/on-call-schedules/_index.md:29:18 MD052/reference-links-images Reference links and images should use a label that is defined [Missing link or image reference definition: "web-schedule"] [Context: "[Web-based schedules][web-schedule]"]
./docs/sources/escalation-chains-and-routes/_index.md:14:18 MD052/reference-links-images Reference links and images should use a label that is defined [Missing link or image reference definition: "routing-template"] [Context: "["Routing Templates"][routing-template]"]
./docs/sources/on-call-schedules/_index.md:29:18 MD052/reference-links-images Reference links and images should use a label that is defined [Missing link or image reference definition: "web-schedule"] [Context: "[Web-based schedules][web-schedule]"]
docs/sources/escalation-chains-and-routes/_index.md:14:18 MD052/reference-links-images Reference links and images should use a label that is defined [Missing link or image reference definition: "routing-template"] [Context: "["Routing Templates"][routing-template]"]
docs/sources/on-call-schedules/_index.md:29:18 MD052/reference-links-images Reference links and images should use a label that is defined [Missing link or image reference definition: "web-schedule"] [Context: "[Web-based schedules][web-schedule]"]
```

imo, I would consider disabling rules that you don't think make sense for the docs. We have a `/docs` specific `.markdownlint.json` [here](https://github.com/grafana/oncall/blob/dev/docs/.markdownlint.json). Hugo !== markdown, so I think getting it to obey by the syntactical "rules" of markdown don't make sense in all cases.

**Comment by jdbaldry** [comment]
Absolutely agree that the problem here is Hugo != Markdown. I really appreciate you recognizing and confirming that.

I'm happy to relax the rules to fix up the CI and will give you another shout when this is ready to go.

**Comment by jdbaldry** [comment]
Turns out the MD052 errors are false positives. I think the shortcode is confusing the parser in some cases.

I still need to work out the diff errors to get this mergeable and fix all the links that are behaving badly in the production website build.

**Comment by jdbaldry** [comment]
I don't understand the MD033 errors in CI. I followed the instructions in https://github.com/DavidAnson/markdownlint/tree/main#configuration to disable that rule for a specific section of the file in https://github.com/grafana/oncall/pull/2456/files#diff-885722bddf5b70461db4e3b67474d96e8e0e3aafa781f7c3005cc7a38fa91f63R195. 

I wonder if I'm doing something wrong?

**Comment by jdbaldry** [comment]
Nice, I disabled some more rules and now CI is happy. I think all disabled rules are necessary to support Hugo syntax. It's a small regression in linting coverage to fix a lot of links.

**Review by joeyorlando — COMMENTED** [review]
overall LGTM 👍 

It's mentioning that the `.github/CODEOWNERS` file contains errors

**Comment by jdbaldry** [comment]
You're right about the CODEOWNERS. I believe it is that the @docs-oncall team has been renamed @docs-gops. 
If you are happy with that change, I'll open a subsequent PR since it is not related to these changes.

File: .github/workflows/update-make-docs.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          BRANCH="$(date +%Y-%m-%d)/update-make-docs"
+          git checkout -b "${BRANCH}"
+          git add .
+          git config --local user.email "bot@grafana.com"
+          git config --local user.name "grafanabot"
+          git commit -m "Update \`make docs\` procedure"
+          git push -v origin "refs/heads/${BRANCH}"
+          gh pr create --fill
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (joeyorlando) [inline .github/workflows/update-make-docs.yml:27]: what does this do?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -107,5 +107,5 @@ repos:
         name: markdownlint
         entry: markdownlint --fix --ignore grafana-plugin/node_modules --ignore grafana-plugin/dist --ignore docs **/*.md
       - id: markdownlint
-        name: markdownlint - docs
-        entry: markdownlint --fix -c ./docs/.markdownlint.json ./docs/**/*.md
+        name: markdownlint - docs/sources
+        entry: markdownlint --fix --ignore README.md -c ./docs/.markdownlint.json ./docs/sources/**/*.md
```
Comment (joeyorlando) [inline .pre-commit-config.yaml:111]: I don't think `README.md` should have to be ignore'd here. `./docs/sources/**/*.md` is telling `markdownlint` to only consider files in that directory

File: .github/workflows/update-make-docs.yml (hunk 1 above)
Comment (jdbaldry) [reply .github/workflows/update-make-docs.yml:27]: It updates the `docs/docs.mk` and `docs/make-docs` scripts that support the `make docs` procedure from a centralized source in Writers' Toolkit every day at 07:00 UTC and opens a PR if there are changes.

**Review by joeyorlando — COMMENTED** [review]
I'm not very familiar w/ the Make changes in `./docs` (or the docs changes themselves). I left a few minor comments on some of the other parts.

It might make sense to get someone from @grafana/docs-gops to approve the rest

File: .pre-commit-config.yaml (hunk 2 above)
Comment (jdbaldry) [reply .pre-commit-config.yaml:111]: Yeah it's weird but without this additional change, `markdownlint` was linting `README.md` and finding errors. I didn't want to spend any more time working out why so I just ignored the file specifically.

File: .github/workflows/update-make-docs.yml (hunk 1 above)
Comment (joeyorlando) [reply .github/workflows/update-make-docs.yml:27]: 👍

File: .pre-commit-config.yaml (hunk 2 above)
Comment (joeyorlando) [reply .pre-commit-config.yaml:111]: 👍

**Review by alyssawada — APPROVED** [review]
From what I can tell, LGTM!

**Pull request merged**